### PR TITLE
Added support for common Scala types (fixes #35)

### LINKED ISSUE #35: Add support for serializing `Byte`, `Short`, `Seq`, and `scala.collection.Map` types
They are not currently supported. But if https://groups.google.com/forum/#!topic/protobuf/Er39mNGnRWU is correct, we should be able to store them as `int32`.

### PR DESCRIPTION
**Pull request by akara**
Definitely did not cover them all, but did cover the basic value types, `Seq`, and `scala.collection.Map` (which does not fit properly into `Map`).

### CODE REVIEW

**Comment by btlines** [comment]
Thanks for the contribution. One minor question/comment otherwise looks good to me.

File: src/main/scala/pbdirect/PBReader.scala [hunk 1]
```diff
@@ -159,6 +167,14 @@ trait PBParserImplicits extends LowPriorityPBParserImplicits {
     instance { (index: Int, bytes: Array[Byte]) =>
       parser.parse(index, bytes).toMap
     }
+  implicit def collectionMapParser[K, V](implicit parser: PBParser[List[(K, V)]]): PBParser[collection.Map[K, V]] =
```
Comment (btlines) [inline src/main/scala/pbdirect/PBReader.scala:170]: what's the need for this? Isn't the `Map` parser good enough if you `import scala.collection.Map`?

Comment (akara) [reply src/main/scala/pbdirect/PBReader.scala:170]: In predef.scala:  `type Map[A, +B] = immutable.Map[A, B]`

And, `immutable.Map` extends `scala.collection.Map`.

So the `mapWriter` and `mapParser` does not cover the type `scala.collection.Map`

You can try removing this parser and you'll find the test does not compile.

Unfortunately, `collection.Map` is being used (and required) by some use cases of Spark and can't just be replaced with `Map`.

Comment (btlines) [reply src/main/scala/pbdirect/PBReader.scala:170]: ok that makes sense

**Review by btlines — APPROVED** [review]

**Pull request merged**